code/game/objects/structures/tables_racks.dm

```diff
@@ -100,11 +148,44 @@
 
 /obj/structure/table/examine(mob/user)
 	. = ..()
+	if(is_flipped)
+		. += span_notice("It's been flipped on its side!")
 	. += deconstruction_hints(user)
 
 /obj/structure/table/proc/deconstruction_hints(mob/user)
 	return span_notice("The top is <b>screwed</b> on, but the main <b>bolts</b> are also visible.")
 
+/obj/structure/table/proc/on_exit(datum/source, atom/movable/leaving, direction)
+	SIGNAL_HANDLER
+	if(!is_flipped)
+		return
+	if(type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(leaving) && (leaving.pass_flags & PASSGLASS))
+			return
```

Comment:
A) Should be istype
B) Would prefer if you unsnowflaked this and made this a param on tables instead